### ISSUE TITLE
v2 dman-v3 script on binance spot - delay executing close order cause of "Account has insufficient balance for requested action" message handling

### ISSUE THREAD

**Issue by nikspz** [hummingbot/hummingbot#6612]
### Describe the bug

v2_mm_dman_v3.py script:
after trailing stop/ time limit order to close activated bot flooded with HTTP status is 400. Error: {"code":-2010,"msg":"Account has insufficient balance for requested action."}

### Steps to reproduce

1. modify v2_mm_dman_v3.py script to use binance spot
2. connect binance
3. use balance 50% for base and 50% for quote token
4. use exactly the balance that used in the script (like 2 buy and 2 sell order levels of 12 usdt each ~ have 49.5 usdt balance total)
5. ran bot for 24 hours or more
6. review logs after order opened

### Release version

dev-1.21.0

### Type of installation

Source

### Attach required files

[logs_v2_market-making_dman_v3_matic_multi_3rd.log](https://github.com/hummingbot/hummingbot/files/12951341/logs_v2_market-making_dman_v3_matic_multi_3rd.log)
![insufficient balance client try to place order to close3](https://github.com/hummingbot/hummingbot/assets/83953535/9267bbb8-6b7d-4efd-80f1-67ad63e96d83)

```
2023-10-17 19:25:53,933 - 3646522 - hummingbot.smart_components.executors.position_executor.position_executor - INFO - Trailing stop activated at 0.51820000
2023-10-17 19:26:06,843 - 3646522 - hummingbot.strategy.script_strategy_base - INFO - Creating MATIC-USDT sell order: price: NaN amount: 22.10000000.
2023-10-17 19:26:06,844 - 3646522 - hummingbot.smart_components.executors.position_executor.position_executor - INFO - Placing close order
2023-10-17 19:26:07,040 - 3646522 - hummingbot.connector.exchange.binance.binance_exchange.BinanceExchange - NETWORK - Error submitting sell MARKET order to Binance for 22.10000000 MATIC-USDT NaN.
Traceback (most recent call last):
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 452, in _create_order
    await self._place_order_and_process_update(order=order, **kwargs,)
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 469, in _place_order_and_process_update
    exchange_order_id, update_timestamp = await self._place_order(
  File "/home/nikita/development/hummingbot/connector/exchange/binance/binance_exchange.py", line 198, in _place_order
    order_result = await self._api_post(
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 867, in _api_post
    return await self._api_request(*args, **kwargs)
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 905, in _api_request
    request_result = await rest_assistant.execute_request(
  File "/home/nikita/development/hummingbot/core/web_assistant/rest_assistant.py", line 47, in execute_request
    response = await self.execute_request_and_get_response(
  File "/home/nikita/development/hummingbot/core/web_assistant/rest_assistant.py", line 99, in execute_request_and_get_response
    raise IOError(f"Error executing request {method.name} {url}. HTTP status is {response.status}. "
OSError: Error executing request POST https://api.binance.com/api/v3/order. HTTP status is 400. Error: {"code":-2010,"msg":"Account has insufficient balance for requested action."}
2023-10-17 19:26:07,041 - 3646522 - hummingbot.connector.exchange.binance.binance_exchange.BinanceExchange - WARNING - Failed to submit SELL order to Binance. Check API key and network connection.
2023-10-17 19:26:07,044 - 3646522 - hummingbot.strategy.script_strategy_base - INFO - Creating MATIC-USDT sell order: price: NaN amount: 22.10000000.
2023-10-17 19:26:07,063 - 3646522 - hummingbot.smart_components.executors.position_executor.position_executor - INFO - Placing close order
2023-10-17 19:26:07,072 - 3646522 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1697570767.0, "order_id": "x-XEKWYICXSMCUT607ee7e89f19142ad", "order_type": "OrderType.MARKET", "event_name": "MarketOrderFailureEvent", "event_source": "binance"}
2023-10-17 19:26:07,076 - 3646522 - hummingbot.connector.client_order_tracker - INFO - Order x-XEKWYICXSMCUT607ee7e89f19142ad has failed. Order Update: OrderUpdate(trading_pair='MATIC-USDT', update_timestamp=1697570767.0, new_state=<OrderState.FAILED: 6>, client_order_id='x-XEKWYICXSMCUT607ee7e89f19142ad', exchange_order_id=None, misc_updates=None)
2023-10-17 19:26:07,278 - 3646522 - hummingbot.connector.exchange.binance.binance_exchange.BinanceExchange - NETWORK - Error submitting sell MARKET order to Binance for 22.10000000 MATIC-USDT NaN.
Traceback (most recent call last):
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 452, in _create_order
    await self._place_order_and_process_update(order=order, **kwargs,)
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 469, in _place_order_and_process_update
    exchange_order_id, update_timestamp = await self._place_order(
  File "/home/nikita/development/hummingbot/connector/exchange/binance/binance_exchange.py", line 198, in _place_order
    order_result = await self._api_post(
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 867, in _api_post
    return await self._api_request(*args, **kwargs)
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 905, in _api_request
    request_result = await rest_assistant.execute_request(
  File "/home/nikita/development/hummingbot/core/web_assistant/rest_assistant.py", line 47, in execute_request
    response = await self.execute_request_and_get_response(
  File "/home/nikita/development/hummingbot/core/web_assistant/rest_assistant.py", line 99, in execute_request_and_get_response
    raise IOError(f"Error executing request {method.name} {url}. HTTP status is {response.status}. "
OSError: Error executing request POST https://api.binance.com/api/v3/order. HTTP status is 400. Error: {"code":-2010,"msg":"Account has insufficient balance for requested action."}
2023-10-17 19:26:07,279 - 3646522 - hummingbot.connector.exchange.binance.binance_exchange.BinanceExchange - WARNING - Failed to submit SELL order to Binance. Check API key and network connection.
2023-10-17 19:26:07,281 - 3646522 - hummingbot.strategy.script_strategy_base - INFO - Creating MATIC-USDT sell order: price: NaN amount: 22.10000000.
2023-10-17 19:26:07,282 - 3646522 - hummingbot.smart_components.executors.position_executor.position_executor - INFO - Placing close order
2023-10-17 19:26:07,288 - 3646522 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1697570767.0, "order_id": "x-XEKWYICXSMCUT607ee7e8cfed442ad", "order_type": "OrderType.MARKET", "event_name": "MarketOrderFailureEvent", "event_source": "binance"}
2023-10-17 19:26:07,291 - 3646522 - hummingbot.connector.client_order_tracker - INFO - Order x-XEKWYICXSMCUT607ee7e8cfed442ad has failed. Order Update: OrderUpdate(trading_pair='MATIC-USDT', update_timestamp=1697570767.0, new_state=<OrderState.FAILED: 6>, client_order_id='x-XEKWYICXSMCUT607ee7e8cfed442ad', exchange_order_id=None, misc_updates=None)
2023-10-17 19:26:07,542 - 3646522 - hummingbot.connector.exchange.binance.binance_exchange.BinanceExchange - NETWORK - Error submitting sell MARKET order to Binance for 22.10000000 MATIC-USDT NaN.
Traceback (most recent call last):
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 452, in _create_order
    await self._place_order_and_process_update(order=order, **kwargs,)
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 469, in _place_order_and_process_update
    exchange_order_id, update_timestamp = await self._place_order(
  File "/home/nikita/development/hummingbot/connector/exchange/binance/binance_exchange.py", line 198, in _place_order
    order_result = await self._api_post(
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 867, in _api_post
    return await self._api_request(*args, **kwargs)
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 905, in _api_request
    request_result = await rest_assistant.execute_request(
  File "/home/nikita/development/hummingbot/core/web_assistant/rest_assistant.py", line 47, in execute_request
    response = await self.execute_request_and_get_response(
  File "/home/nikita/development/hummingbot/core/web_assistant/rest_assistant.py", line 99, in execute_request_and_get_response
    raise IOError(f"Error executing request {method.name} {url}. HTTP status is {response.status}. "
OSError: Error executing request POST https://api.binance.com/api/v3/order. HTTP status is 400. Error: {"code":-2010,"msg":"Account has insufficient balance for requested action."}
2023-10-17 19:26:07,543 - 3646522 - hummingbot.connector.exchange.binance.binance_exchange.BinanceExchange - WARNING - Failed to submit SELL order to Binance. Check API key and network connection.
2023-10-17 19:26:07,631 - 3646522 - hummingbot.strategy.script_strategy_base - INFO - Creating MATIC-USDT sell order: price: NaN amount: 22.10000000.
2023-10-17 19:26:07,632 - 3646522 - hummingbot.smart_components.executors.position_executor.position_executor - INFO - Placing close order
2023-10-17 19:26:07,638 - 3646522 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1697570767.0, "order_id": "x-XEKWYICXSMCUT607ee7e90a0a442ad", "order_type": "OrderType.MARKET", "event_name": "MarketOrderFailureEvent", "event_source": "binance"}
2023-10-17 19:26:07,641 - 3646522 - hummingbot.connector.client_order_tracker - INFO - Order x-XEKWYICXSMCUT607ee7e90a0a442ad has failed. Order Update: OrderUpdate(trading_pair='MATIC-USDT', update_timestamp=1697570767.0, new_state=<OrderState.FAILED: 6>, client_order_id='x-XEKWYICXSMCUT607ee7e90a0a442ad', exchange_order_id=None, misc_updates=None)
2023-10-17 19:26:07,916 - 3646522 - hummingbot.connector.exchange.binance.binance_exchange.BinanceExchange - NETWORK - Error submitting sell MARKET order to Binance for 22.10000000 MATIC-USDT NaN.
Traceback (most recent call last):
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 452, in _create_order
    await self._place_order_and_process_update(order=order, **kwargs,)
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 469, in _place_order_and_process_update
    exchange_order_id, update_timestamp = await self._place_order(
  File "/home/nikita/development/hummingbot/connector/exchange/binance/binance_exchange.py", line 198, in _place_order
    order_result = await self._api_post(
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 867, in _api_post
    return await self._api_request(*args, **kwargs)
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 905, in _api_request
    request_result = await rest_assistant.execute_request(
  File "/home/nikita/development/hummingbot/core/web_assistant/rest_assistant.py", line 47, in execute_request
    response = await self.execute_request_and_get_response(
  File "/home/nikita/development/hummingbot/core/web_assistant/rest_assistant.py", line 99, in execute_request_and_get_response
    raise IOError(f"Error executing request {method.name} {url}. HTTP status is {response.status}. "
OSError: Error executing request POST https://api.binance.com/api/v3/order. HTTP status is 400. Error: {"code":-2010,"msg":"Account has insufficient balance for requested action."}
2023-10-17 19:26:07,916 - 3646522 - hummingbot.connector.exchange.binance.binance_exchange.BinanceExchange - WARNING - Failed to submit SELL order to Binance. Check API key and network connection.
2023-10-17 19:26:07,918 - 3646522 - hummingbot.strategy.script_strategy_base - INFO - Creating MATIC-USDT sell order: price: NaN amount: 22.10000000.
2023-10-17 19:26:07,919 - 3646522 - hummingbot.smart_components.executors.position_executor.position_executor - INFO - Placing close order
2023-10-17 19:26:07,925 - 3646522 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1697570767.0, "order_id": "x-XEKWYICXSMCUT607ee7e95f69a42ad", "order_type": "OrderType.MARKET", "event_name": "MarketOrderFailureEvent", "event_source": "binance"}
2023-10-17 19:26:07,928 - 3646522 - hummingbot.connector.client_order_tracker - INFO - Order x-XEKWYICXSMCUT607ee7e95f69a42ad has failed. Order Update: OrderUpdate(trading_pair='MATIC-USDT', update_timestamp=1697570767.0, new_state=<OrderState.FAILED: 6>, client_order_id='x-XEKWYICXSMCUT607ee7e95f69a42ad', exchange_order_id=None, misc_updates=None)
2023-10-17 19:26:08,209 - 3646522 - hummingbot.connector.exchange.binance.binance_exchange.BinanceExchange - NETWORK - Error submitting sell MARKET order to Binance for 22.10000000 MATIC-USDT NaN.
Traceback (most recent call last):
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 452, in _create_order
    await self._place_order_and_process_update(order=order, **kwargs,)
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 469, in _place_order_and_process_update
    exchange_order_id, update_timestamp = await self._place_order(
  File "/home/nikita/development/hummingbot/connector/exchange/binance/binance_exchange.py", line 198, in _place_order
    order_result = await self._api_post(
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 867, in _api_post
    return await self._api_request(*args, **kwargs)
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 905, in _api_request
    request_result = await rest_assistant.execute_request(
  File "/home/nikita/development/hummingbot/core/web_assistant/rest_assistant.py", line 47, in execute_request
    response = await self.execute_request_and_get_response(
  File "/home/nikita/development/hummingbot/core/web_assistant/rest_assistant.py", line 99, in execute_request_and_get_response
    raise IOError(f"Error executing request {method.name} {url}. HTTP status is {response.status}. "
OSError: Error executing request POST https://api.binance.com/api/v3/order. HTTP status is 400. Error: {"code":-2010,"msg":"Account has insufficient balance for requested action."}
2023-10-17 19:26:08,209 - 3646522 - hummingbot.connector.exchange.binance.binance_exchange.BinanceExchange - WARNING - Failed to submit SELL order to Binance. Check API key and network connection.
2023-10-17 19:26:08,212 - 3646522 - hummingbot.strategy.script_strategy_base - INFO - Creating MATIC-USDT sell order: price: NaN amount: 22.10000000.
2023-10-17 19:26:08,213 - 3646522 - hummingbot.smart_components.executors.position_executor.position_executor - INFO - Placing close order
2023-10-17 19:26:08,219 - 3646522 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1697570768.0, "order_id": "x-XEKWYICXSMCUT607ee7e9a59ed42ad", "order_type": "OrderType.MARKET", "event_name": "MarketOrderFailureEvent", "event_source": "binance"}
2023-10-17 19:26:08,221 - 3646522 - hummingbot.connector.client_order_tracker - INFO - Order x-XEKWYICXSMCUT607ee7e9a59ed42ad has failed. Order Update: OrderUpdate(trading_pair='MATIC-USDT', update_timestamp=1697570768.0, new_state=<OrderState.FAILED: 6>, client_order_id='x-XEKWYICXSMCUT607ee7e9a59ed42ad', exchange_order_id=None, misc_updates=None)


........................................................................... Events of failed to place order every seconds for ~90 sec




2023-10-17 19:27:41,216 - 3646522 - hummingbot.connector.exchange.binance.binance_exchange.BinanceExchange - NETWORK - Error submitting sell MARKET order to Binance for 22.10000000 MATIC-USDT NaN.
Traceback (most recent call last):
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 452, in _create_order
    await self._place_order_and_process_update(order=order, **kwargs,)
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 469, in _place_order_and_process_update
    exchange_order_id, update_timestamp = await self._place_order(
  File "/home/nikita/development/hummingbot/connector/exchange/binance/binance_exchange.py", line 198, in _place_order
    order_result = await self._api_post(
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 867, in _api_post
    return await self._api_request(*args, **kwargs)
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 905, in _api_request
    request_result = await rest_assistant.execute_request(
  File "/home/nikita/development/hummingbot/core/web_assistant/rest_assistant.py", line 47, in execute_request
    response = await self.execute_request_and_get_response(
  File "/home/nikita/development/hummingbot/core/web_assistant/rest_assistant.py", line 99, in execute_request_and_get_response
    raise IOError(f"Error executing request {method.name} {url}. HTTP status is {response.status}. "
OSError: Error executing request POST https://api.binance.com/api/v3/order. HTTP status is 400. Error: {"code":-2010,"msg":"Account has insufficient balance for requested action."}
2023-10-17 19:27:41,217 - 3646522 - hummingbot.connector.exchange.binance.binance_exchange.BinanceExchange - WARNING - Failed to submit SELL order to Binance. Check API key and network connection.
2023-10-17 19:27:41,219 - 3646522 - hummingbot.strategy.script_strategy_base - INFO - Creating MATIC-USDT sell order: price: NaN amount: 22.10000000.
2023-10-17 19:27:41,220 - 3646522 - hummingbot.smart_components.executors.position_executor.position_executor - INFO - Placing close order
2023-10-17 19:27:41,226 - 3646522 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1697570861.0, "order_id": "x-XEKWYICXSMCUT607ee842591ee42ad", "order_type": "OrderType.MARKET", "event_name": "MarketOrderFailureEvent", "event_source": "binance"}
2023-10-17 19:27:41,229 - 3646522 - hummingbot.connector.client_order_tracker - INFO - Order x-XEKWYICXSMCUT607ee842591ee42ad has failed. Order Update: OrderUpdate(trading_pair='MATIC-USDT', update_timestamp=1697570861.0, new_state=<OrderState.FAILED: 6>, client_order_id='x-XEKWYICXSMCUT607ee842591ee42ad', exchange_order_id=None, misc_updates=None)
2023-10-17 19:27:41,430 - 3646522 - hummingbot.connector.exchange.binance.binance_exchange.BinanceExchange - NETWORK - Error submitting sell MARKET order to Binance for 22.10000000 MATIC-USDT NaN.
Traceback (most recent call last):
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 452, in _create_order
    await self._place_order_and_process_update(order=order, **kwargs,)
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 469, in _place_order_and_process_update
    exchange_order_id, update_timestamp = await self._place_order(
  File "/home/nikita/development/hummingbot/connector/exchange/binance/binance_exchange.py", line 198, in _place_order
    order_result = await self._api_post(
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 867, in _api_post
    return await self._api_request(*args, **kwargs)
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 905, in _api_request
    request_result = await rest_assistant.execute_request(
  File "/home/nikita/development/hummingbot/core/web_assistant/rest_assistant.py", line 47, in execute_request
    response = await self.execute_request_and_get_response(
  File "/home/nikita/development/hummingbot/core/web_assistant/rest_assistant.py", line 99, in execute_request_and_get_response
    raise IOError(f"Error executing request {method.name} {url}. HTTP status is {response.status}. "
OSError: Error executing request POST https://api.binance.com/api/v3/order. HTTP status is 400. Error: {"code":-2010,"msg":"Account has insufficient balance for requested action."}
2023-10-17 19:27:41,431 - 3646522 - hummingbot.connector.exchange.binance.binance_exchange.BinanceExchange - WARNING - Failed to submit SELL order to Binance. Check API key and network connection.
2023-10-17 19:27:41,441 - 3646522 - hummingbot.strategy.script_strategy_base - INFO - Creating MATIC-USDT sell order: price: NaN amount: 22.10000000.
2023-10-17 19:27:41,467 - 3646522 - hummingbot.smart_components.executors.position_executor.position_executor - INFO - Placing close order
2023-10-17 19:27:41,473 - 3646522 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1697570861.0, "order_id": "x-XEKWYICXSMCUT607ee842a032842ad", "order_type": "OrderType.MARKET", "event_name": "MarketOrderFailureEvent", "event_source": "binance"}
2023-10-17 19:27:41,477 - 3646522 - hummingbot.connector.client_order_tracker - INFO - Order x-XEKWYICXSMCUT607ee842a032842ad has failed. Order Update: OrderUpdate(trading_pair='MATIC-USDT', update_timestamp=1697570861.0, new_state=<OrderState.FAILED: 6>, client_order_id='x-XEKWYICXSMCUT607ee842a032842ad', exchange_order_id=None, misc_updates=None)
2023-10-17 19:27:41,478 - 3646522 - hummingbot.strategy.script_strategy_base - INFO - (MATIC-USDT) Canceling the limit order x-XEKWYICXSMCUT607ee7242858642ad. [clock=2023-10-17 19:27:41+00:00]
2023-10-17 19:27:41,943 - 3646522 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1697570861.0, "order_id": "x-XEKWYICXSMCUT607ee7242858642ad", "exchange_order_id": "3078300166", "event_name": "OrderCancelledEvent", "event_source": "binance"}
2023-10-17 19:27:41,945 - 3646522 - hummingbot.connector.client_order_tracker - INFO - Successfully canceled order x-XEKWYICXSMCUT607ee7242858642ad.
2023-10-17 19:27:41,960 - 3646522 - hummingbot.connector.client_order_tracker - INFO - The SELL order x-XEKWYICXSMCUT607ee842dc65b42ad amounting to 22.10000000/22.10000000 MATIC has been filled.
```

**Comment by cardosofede** [comment]
@nikspz DManV3 has a bug in the controller that I just fixed in the branch of OrderLevelBuilder.
The value of time limit was multiplied by the target spread.
Regarding to this issue, you have the orders placed so it's natural that you can't close the position. Do the following to validate the behavior of this:
- Hold 50 USDT and 50 (USD) in MATIC
- Create a DMAN V3 strategy using the order level builder branch with 2 order levels of 12 usd each
- Evaluate if this happens again

**Comment by nikspz** [comment]
fixed in #6615
